pkg/kubelet/config/file_test.go

```diff
@@ -127,109 +73,152 @@ func TestReadManifestFromFile(t *testing.T) {
-			ObjectMeta: api.ObjectMeta{
-				Name:      "test-" + hostname,
-				UID:       "12345",
-				Namespace: kubelet.NamespaceDefault,
-				SelfLink:  "/api/v1beta2/pods/test-" + hostname + "?namespace=default",
-			},
-			Spec: api.PodSpec{Containers: []api.Container{{Image: "test/image"}}},
-		})
-
-		if !api.Semantic.DeepDerivative(expected, update) {
-			t.Fatalf("Expected %#v, Got %#v", expected, update)
-		}
-
-	case <-time.After(time.Second):
-		t.Errorf("Expected update, timeout instead")
-	}
-}
-
-func TestReadManifestFromFileWithoutID(t *testing.T) {
-	hostname, _ := os.Hostname()
-	hostname = strings.ToLower(hostname)
-
-	file := writeTestFile(t, os.TempDir(), "test_pod_config",
-		`{
-			"version": "v1beta1",
-			"uuid": "12345",
-			"containers": [{ "name": "image", "image": "test/image", imagePullPolicy: "PullAlways"}]
-		}`)
-	defer os.Remove(file.Name())
-
-	ch := make(chan interface{})
-	NewSourceFile(file.Name(), time.Millisecond, ch)
-	select {
-	case got := <-ch:
-		update := got.(kubelet.PodUpdate)
-		expected := CreatePodUpdate(kubelet.SET, kubelet.FileSource, api.Pod{
-			ObjectMeta: api.ObjectMeta{
-				Name:      "12345-" + hostname,
-				UID:       "12345",
-				Namespace: kubelet.NamespaceDefault,
-				SelfLink:  "/api/v1beta2/pods/12345-" + hostname + "?namespace=default",
-			},
-			Spec: api.PodSpec{Containers: []api.Container{{Image: "test/image"}}},
-		})
-
-		if !api.Semantic.DeepDerivative(expected, update) {
-			t.Fatalf("Expected %#v, Got %#v", expected, update)
-		}
-
-	case <-time.After(time.Second):
-		t.Errorf("Expected update, timeout instead")
+	var testCases = []struct {
+		desc         string
+		fileContents string
+		expected     kubelet.PodUpdate
+	}{
+		{
+			desc: "Manifest",
+			fileContents: `{
+					"version": "v1beta1",
+					"uuid": "12345",
+					"id": "test",
+					"containers": [{ "name": "image", "image": "test/image", "imagePullPolicy": "PullAlways"}]
+				}`,
+			expected: CreatePodUpdate(kubelet.SET, kubelet.FileSource, api.Pod{
+				ObjectMeta: api.ObjectMeta{
+					Name:      "test-" + hostname,
+					UID:       "12345",
+					Namespace: kubelet.NamespaceDefault,
+					SelfLink:  "/api/v1beta2/pods/test-" + hostname + "?namespace=default",
+				},
+				Spec: api.PodSpec{Containers: []api.Container{{Image: "test/image"}}},
+			}),
+		},
+		{
+			desc: "Manifest without ID",
+			fileContents: `{
+					"version": "v1beta1",
+					"uuid": "12345",
+					"containers": [{ "name": "image", "image": "test/image", "imagePullPolicy": "PullAlways"}]
+				}`,
+			expected: CreatePodUpdate(kubelet.SET, kubelet.FileSource, api.Pod{
+				ObjectMeta: api.ObjectMeta{
+					Name:      "12345-" + hostname,
+					UID:       "12345",
+					Namespace: kubelet.NamespaceDefault,
+					SelfLink:  "/api/v1beta2/pods/12345-" + hostname + "?namespace=default",
+				},
+				Spec: api.PodSpec{Containers: []api.Container{{Image: "test/image"}}},
```

Comment:
Shouldn't these have the container name and imagePullPolicy? I'm a bit confused the tests pass. We do include these in http_test.go